pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -0,0 +1,472 @@
+from pytorch_lightning.plugins.environments.torchelastic_environment import TorchElasticEnvironment
+from pytorch_lightning.tuner.auto_gpu_select import pick_multiple_gpus
+from pytorch_lightning.utilities import (
+    _APEX_AVAILABLE,
+    _HOROVOD_AVAILABLE,
+    _NATIVE_AMP_AVAILABLE,
+    _TPU_AVAILABLE,
+    AMPType,
+    device_parser,
+    DeviceType,
+    DistributedType,
+    rank_zero_only,
+)
+from pytorch_lightning.utilities.distributed import rank_zero_info, rank_zero_warn
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+if _HOROVOD_AVAILABLE:
+    import horovod.torch as hvd
+
+
+class BackendConnector(object):
+    def __init__(
+        self,
+        num_processes,
+        tpu_cores,
+        distributed_backend,
+        auto_select_gpus,
+        gpus,
+        num_nodes,
+        sync_batchnorm,
+        benchmark,
+        replace_sampler_ddp,
+        deterministic,
+        precision,
+        amp_type,
+        amp_level,
+        cluster_environment,
+    ):
+        # initialization
+        self._device_type = DeviceType.CPU
+        self._distrib_type = None
+
+        self.num_processes = num_processes
+        self.tpu_cores = device_parser.parse_tpu_cores(tpu_cores)
+        self.distributed_backend = distributed_backend
+        self.auto_select_gpus = auto_select_gpus
+        self.gpus = gpus
+        self.num_nodes = num_nodes
+        self.sync_batchnorm = sync_batchnorm
+        self.benchmark = benchmark
+        self.replace_sampler_ddp = replace_sampler_ddp
+        self.deterministic = deterministic
+        self.precision = precision
+        self.amp_type = None if amp_type is None else amp_type.lower()
+        self.amp_level = amp_level
+        self.cluster_environment = cluster_environment
+        self.is_slurm_managing_tasks = False
+
+        # init the default rank if exists
+        # we need to call this here or NVIDIA flags and other messaging in init will show on all ranks
+        # this way we only show it on rank 0
+        if "LOCAL_RANK" in os.environ:
+            rank_zero_only.rank = int(os.environ["LOCAL_RANK"])
+
+        # for gpus allow int, string and gpu list
+        if auto_select_gpus and isinstance(gpus, int):
+            self.gpus = pick_multiple_gpus(gpus)
+
+        self.parallel_device_ids = device_parser.parse_gpu_ids(self.gpus)
+        self.root_gpu = device_parser.determine_root_gpu_device(self.parallel_device_ids)
+
+        self.set_distributed_mode()
+        self.configure_slurm_ddp()
+
+        self.accelerator = self.select_accelerator()
+
+        # override dist backend when using tpus
+        if self.on_tpu:
+            self.distributed_backend = "tpu"
+            self.use_tpu = True
+
+        # init flags for SLURM+DDP to work
+        self.world_size = 1
+        self.interactive_ddp_procs = []
+        self.global_rank = 0
+
+        # NVIDIA setup
```

Comment:
This should be moved to GPU Accelerator.

Reply:
the nivida setup, yes definitely. 
I will work on this on the PoC #5616 branch asap